charts/kube-prometheus-stack/templates/prometheus-operator/deployment.yaml

```diff
@@ -91,10 +91,26 @@ spec:
             {{- else }}
             - --prometheus-config-reloader={{ $configReloaderRegistry }}/{{ .Values.prometheusOperator.prometheusConfigReloader.image.repository }}:{{ .Values.prometheusOperator.prometheusConfigReloader.image.tag | default .Chart.AppVersion }}
             {{- end }}
+            {{- if .Values.prometheusOperator.prometheusConfigReloader.resources.requests.cpu }}
             - --config-reloader-cpu-request={{ .Values.prometheusOperator.prometheusConfigReloader.resources.requests.cpu }}
+            {{- else }}
+            - --config-reloader-cpu-request=0
+            {{- end }}
```

Comment:
This is a good take but still not there as one gets a _nil pointer evaluating interface {}.cpu_ error as before. I can suggest two ways of tackling this:
1. This is very similar but using "pipelines" `()` enclosing a nullable level, e.g.
```
{{- if (((.Values.prometheusOperator.prometheusConfigReloader.resources).requests).cpu) }}
- --config-reloader-cpu-request={{ .Values.prometheusOperator.prometheusConfigReloader.resources.requests.cpu }}
{{- else }}
- --config-reloader-cpu-request=0
{{- end }}
```

2. Using `dig` function returning a default (0 in this case) if any of the nested fields `requests` and `cpu` (or both) is absent from `.Values.prometheusOperator.prometheusConfigReloader.resources`, e.g.

```
{{- dig "requests" "cpu" 0 .Values.prometheusOperator.prometheusConfigReloader.resources }}
```
The function has been available in helm since 3.5.0.

Reply:
Thanks again @zeritti
I updated it accordingly